packages/aws-cdk-lib/aws-sns-subscriptions/lib/subscription.ts

```diff
@@ -24,5 +28,60 @@ export interface SubscriptionProps {
    *
    * @default - No dead letter queue enabled.
    */
-  readonly deadLetterQueue?: IQueue;
+  readonly deadLetterQueue?: sqs.IQueue;
+}
+
+/**
+ * Generic subscription target
+ */
+export abstract class Subscription implements sns.ITopicSubscription {
+  constructor(private readonly subscriber: IResource) {
```

Comment:
`Subscription.subscriber` ends up having the same value as the existing `SqsSubscription.queue` and `LambdaSubscription.fn` members, but with a broader typing.

I would have preferred to have a common `protected` property between all 3, and have the class implementations overload the abstract class's `IResource`, but jsii wouldn't allow it. The alternatives would be to convert `generateGrantPrincipal` into a static method, and pass both the topic and the subscriber as parameters. Neither feel particularly great, but I slightly prefer my current implementation